### PR TITLE
feat: add button-dropdown package

### PR DESCRIPTION
**Pull request by twebber90**
### Feature
Add `@tractorzoom/button-dropdown` package

- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] Any dependent changes have been merged and published in downstream modules
- [X] The test workflow is passing locally

### defaults
Screen Shot 2020-08-03 at 12 37 35 PM

### with label and color
Screen Shot 2020-08-03 at 12 37 45 PM

### CODE REVIEW

**Comment by twebber90** [comment]
Right now it is just taking a generic `onChange` event that is called with the string option that is selected/changed

File: packages/button-dropdown/src/button-dropdown.js [hunk 1]
```diff
@@ -12,14 +12,19 @@ import PropTypes from 'prop-types';
 
 const ButtonDropdown = (props) => {
     const [open, setOpen] = useState(false);
+    const [selectedIndex, setSelectedIndex] = useState(props.options.indexOf(props.selectedOption));
```
Comment (cody-hoffman) [inline packages/button-dropdown/src/button-dropdown.js:15]: Since selectedOption is optional, is there a default selected option or how does this work?

Comment (twebber90) [reply packages/button-dropdown/src/button-dropdown.js:15]: If no `selectedOption` is passed then the `selectedIndex` is null

Comment (cody-hoffman) [reply packages/button-dropdown/src/button-dropdown.js:15]: won't that blow up the onChange then?

Comment (twebber90) [reply packages/button-dropdown/src/button-dropdown.js:15]: Screen Shot 2020-08-03 at 2 06 38 PM

☝️ is what gets fired without an option selected and then with an option selected @cody-hoffman

**Pull request merged**